packages/spanner/src/database.js

```diff
@@ -233,6 +244,12 @@ Database.formatName_ = function(instanceName, name) {
  */
 Database.prototype.close = function(callback) {
   var self = this;
+  var leaks = this.pool_.getSessionLeaks();
+
+  if (leaks.length) {
+    callback(new SessionLeakError(leaks));
+    return;
```

Comment:
That's possible, although currently we do not keep track of leased sessions, but I don't think that would be too difficult to do.